mlmicrophysics/module_neural_net.f90

```diff
@@ -106,30 +106,38 @@ subroutine apply_activation(input, activation_type, output)
                 end do
             case (2)
             !    print*, "sigmoid"
-                output = 1.0 / (1.0 + exp(-input))
+                output = 1.0 / (1.0 + dexp(-input))
```

Comment:
Why change exponential function to exponential density function?

Reply:
dexp is still the element-by-element exponential function but specific to arrays containing double precision floating point numbers. The exp function in fortran is specific to single precision (32 bit) floating point values. Using dexp ensures that round-off errors or type mismatch errors aren't introduced since the rest of module_neural_net assumes 64-bit values for all inputs and weight arrays.